source/common/network/listener_impl.cc

```diff
@@ -23,77 +20,39 @@ Address::InstanceConstSharedPtr ListenerImpl::getLocalAddress(int fd) {
   return Address::addressFromFd(fd);
 }
 
-Address::InstanceConstSharedPtr ListenerImpl::getOriginalDst(int fd) {
-  return Utility::getOriginalDst(fd);
-}
-
 void ListenerImpl::listenCallback(evconnlistener*, evutil_socket_t fd, sockaddr* remote_addr,
                                   int remote_addr_len, void* arg) {
   ListenerImpl* listener = static_cast<ListenerImpl*>(arg);
-  Address::InstanceConstSharedPtr final_local_address = listener->socket_.localAddress();
-  bool using_original_dst = false;
-
-  // Get the local address from the new socket if the listener is listening on the all hosts
-  // address (e.g., 0.0.0.0 for IPv4).
-  auto ip = final_local_address->ip();
-  if (ip && ip->isAnyAddress()) {
-    final_local_address = listener->getLocalAddress(fd);
-  }
-
-  if (listener->options_.use_original_dst_ && final_local_address->type() == Address::Type::Ip) {
-    Address::InstanceConstSharedPtr original_local_address = listener->getOriginalDst(fd);
-
-    // A listener that has the use_original_dst flag set to true can still receive
-    // connections that are NOT redirected using iptables. If a connection was not redirected,
-    // the address returned by getOriginalDst() matches the local address of the new socket.
-    // In this case the listener handles the connection directly and does not hand it off.
-    if (original_local_address && (*original_local_address != *final_local_address)) {
-      final_local_address = original_local_address;
-      using_original_dst = true;
-
-      // Hands off redirected connections (from iptables) to the listener associated with the
-      // original destination address. If there is no listener associated with the original
-      // destination address, the connection is handled by the listener that receives it.
-      ListenerImpl* new_listener = dynamic_cast<ListenerImpl*>(
-          listener->connection_handler_.findListenerByAddress(*original_local_address));
-
-      if (new_listener != nullptr) {
-        listener = new_listener;
-      }
-    }
-  }
-
-  if (listener->options_.use_proxy_proto_) {
-    listener->proxy_protocol_.newConnection(listener->dispatcher_, fd, *listener);
-  } else {
-    Address::InstanceConstSharedPtr final_remote_address;
-    if (remote_addr->sa_family == AF_UNIX) {
+  AcceptedSocketPtr socket(new AcceptedSocketImpl(
+      fd,
+      // Get the local address from the new socket if the listener is listening on the all hosts
+      // address (e.g., 0.0.0.0 for IPv4).
+      !listener->local_address_ ? listener->getLocalAddress(fd) : listener->local_address_,
```

Comment:
So local_address_ is nullptr for 0.0.0.0? Or it's the ANY address? I thought it was the latter? Maybe I just don't remember. Can you add to comment that ANY address is nullptr if that is the case? OK I see below you changed this. Can you just add comment here that nullptr indicates any address?